server/aws-lsp-identity/src/language-server/identityService.ts

```diff
@@ -136,6 +164,66 @@ export class IdentityService {
         }
     }
 
+    async getIamCredential(params: GetIamCredentialParams, token: CancellationToken): Promise<GetIamCredentialResult> {
```

Comment:
The impl for this is likely to get HUGE as more profile types are added.  This impl should be moved under ../iam/ (similar to ../sso/) with the function here being mostly input/output validation, logging, and handing off primary impl to code under that namespace.